### ISSUE TITLE
request: warn when // @vue/component is not followed by an object

### ISSUE THREAD

**Issue by KaelWD** [vuejs/eslint-plugin-vue#516]
* **ESLint Version:** 4.19.1
* **eslint-plugin-vue Version:** 4.5.0
* **Node Version:** 9.11.1

**What did you do? Please include the actual source code causing the issue.**

```js
/* @vue/component */
export default mixins(
  Colorable,
  Toggleable,
  PositionableFactory(['left', 'bottom']),
  Transitionable
).extend({
  props: ['a-prop'],

  name: 'a-component'
```

**What did you expect to happen?**
eslint should warn about `/* @vue/component */` not being followed by a component object, because it needs to be directly above the `.extend({` line to work. 

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing, it passes.

**Comment by michalsnik** [comment]
Hey, thanks for this proposition @KaelWD. I think that if we want to address this matter it would be best to introduce an extra rule that would check and report warning in case the comment is not followed by an object. It could be named `valid-vue-comment-location`.
